xla/service/hlo_verifier.cc

```diff
@@ -214,6 +214,13 @@ Status ShapeVerifier::HandleDot(HloInstruction* dot) {
           dot->operand(0)->shape(), dot->operand(1)->shape(),
           dot->dot_dimension_numbers(),
           /*preferred_element_type=*/dot->shape().element_type(), sparsity));
+  if (!HasCompatibleElementTypes(dot->operand(0)->shape(),
```

Comment:
allow_mixed_precision is about floating point mixed types. But dot allows also other mixed types (e.g. u8 inputs and u32 output), so adding this check here would mean they would suddenly be considered invalid (but codegen works fine for them).

Reply:
Will fix. Expect new review soon

Reply:
Still a lot of tests are failing. Note that allow_mixed_precision in the verifier is set to false by default, and tests usually don't override the default. So all those tests would have to be changed to set allow_mixed_precision to true. Also I don't really see the point if we then set allow_mixed_precision to true everywhere. Then this new check in the verifier will just be dead code?

Reply:
You're right, it would be a lot of work to fix all the tests that rely on the fact that the verifier is broken. However, having the verifier work as intended is important for the customers and stakeholders who use xla. Possible solutions include setting `allow_mixed_precision` to true for the scope of the tests,  setting `allow_mixed_precision` to true globally or in the worst case scenario fixing the errant tests by hand. Could you please let me know which of these is the most/least desirable to you?

Reply:
I would guess setting allow_mixed_precision to true globally might be the best option. But then, the new check in the verifier will never trigger, or do I miss something? So I am not exactly sure why you want this check in the verifier. Essentially, for dot we always allow mixed precision, so why do we need code in the verifier to trigger if we don't allow mixed precision? Do you have a XLA backend where you do not want to allow mixed precision? In that case, the default verifier option should probably be derived from some option that the backend can configure.

Reply:
I do have an XLA backend where I do not want to allow mixed precision. Having the backend set the default verifier option sounds like the best option. Lets do that.